### PR TITLE
ensure alternates collection is valid

### LINKED ISSUE #5285: Orchard.ConsoleLog(shape) producing error.
Following the guide here to show shape in chrome browser :

https://orcharddojo.net/blog/consolelog-helper-issectiondefined-method-this-week-in-orchard-03-01-2020

I am getting the following error : 

image

@deanmarcussen

### PR DESCRIPTION
**Pull request by giannik**
Fixes https://github.com/OrchardCMS/OrchardCore/issues/5285

ensure collection index is valid

### CODE REVIEW

**Comment by deanmarcussen** [comment]
nice one @giannik for following through and finding that :)

**Comment by sebastienros** [comment]
what is this syntax ;) googling ...

**Comment by sebastienros** [comment]
What about `LastOrDefault()` which seems more readable?

**Comment by sebastienros** [comment]
Yeah, I am starting to hate C# https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/ranges

**Comment by Jetski5822** [comment]
FFS I need to learn this too - grrr

File: src/OrchardCore/OrchardCore.DisplayManagement/Shapes/AlternatesCollection.cs [hunk 1]
```diff
@@ -26,7 +26,7 @@ public AlternatesCollection(params string[] alternates)
 
         public string this[int index] => _collection[index];
 
-        public string Last => _collection[_collection.Count - 1];
+        public string Last => _collection.Count >0 ? _collection[^1]: "";
```
Comment (jtkech) [inline src/OrchardCore/OrchardCore.DisplayManagement/Shapes/AlternatesCollection.cs]: If you use our source code with our `.editorconfig` file and under Visual Studio, try to `ctrl K+D`, you will see that it adds a space after the `>`.

I learnt something today about c# 8.0, thanks ;)

**Comment by sebastienros** [comment]
Because I was not allowed to update the PR I pushed the changes directly to dev, thanks

**Pull request closed without merging**